packages/liveblocks-node/src/PermissionSet.ts

```diff
@@ -0,0 +1,146 @@
+  return (ALL_PERMISSIONS as readonly unknown[]).includes(value);
+}
+
+const MAX_PERMS_PER_SET = 10;
+
+/**
+ * Assign this to a room or room pattern if you want to grant the user
+ * read-only permissions for this room.
+ */
+export const READ_ACCESS: readonly Permission[] = Object.freeze([
+  "room:read",
+  // "room:presence:write",  // XXX Should this be included in the READ_ACCESS set?
+  "comments:read",
+]);
+
+/**
+ * Assign this to a room or room pattern if you want to grant the user all
+ * permissions for this room.
+ */
+export const FULL_ACCESS: readonly Permission[] = Object.freeze([
+  "room:write",
+  "comments:write",
+
+  // XXX Are these implied automatically if you have the two write permissions above? If so, it would be nice to keep the JWT payload as small as possible.
+  // "room:read",
+  // "room:presence:write",
+  // "comments:read",
+]);
+
+const roomPatternRegex = /^[^*]{1,50}[*]?$/;
+
+/**
+ * Class to help you construct the exact permission set to grant a user, used
+ * when making `.authorizeUser()` calls.
+ *
+ * Usage:
+ *
+ *    const p = new Permissions();
+ *    p.allow(roomId, scopes)  // or...
+ *
+ * Example:
+ *
+ *    p.allow('my-room', FULL_ACCESS)  // Grant read + write access to 'my-room'
+ *    p.allow('my-room', READ_ACCESS)  // Grant read-only access to 'my-room'
+ *
+ * Rooms can be specified with a prefix match, if the name ends in an asterisk.
+ * In that case, access is granted to all rooms that start with that pattern:
+ *
+ *    p.allow('abc:*', FULL_ACCESS)    // Grant read + write access to all rooms that start with "abc:"
+ *
+ * You can define at most 10 room IDs or patterns in a single token, otherwise
+ * the token would become too large and unwieldy.
+ *
+ * All permissions granted are additive. You cannot "remove" permissions once
+ * you grant them. For example:
+ *
+ *    p
+ *      .allow('abc:*',   FULL_ACCESS)
+ *      .allow('abc:123', READ_ACCESS)
+ *
+ * Here, room `abc:123` would have full access. The second .allow() call only
+ * _adds_ read permissions, but that has no effect since full access
+ * permissions were already added to the set.
+ */
+export class PermissionSet {
+  private _sealed = false;
+  private readonly _permissions: Map<string, Set<Permission>> = new Map();
+
+  private getOrCreate(roomId: string): Set<Permission> {
+    if (this._sealed) {
+      throw new Error("You can no longer change these permissions.");
+    }
+
+    let perms = this._permissions.get(roomId);
+    if (perms) {
+      return perms;
+    } else {
+      if (this._permissions.size >= MAX_PERMS_PER_SET) {
+        throw new Error(
+          "You cannot add permissions for more than 10 rooms in a single token"
+        );
+      }
+
+      perms = new Set<Permission>();
+      this._permissions.set(roomId, perms);
+      return perms;
+    }
+  }
+
+  public allow(
+    roomIdOrPattern: string,
+    newPerms: readonly Permission[] = FULL_ACCESS
```

Comment:
@ofoucherot Also, what do you think of this default (and the [related unit test](https://github.com/liveblocks/liveblocks/blob/5a779cd91cc930fa437bd608422f0bc20a9af451/packages/liveblocks-node/src/__tests__/PermissionSet.test.ts#L77-L89))? Is it convenient to make it the default, or should we remove this and ask people to be explicit about it?

I still kinda dislike that we have to import the presets from the global module, i.e.

```ts
import { FULL_ACCESS } from ...
```

Perhaps we can put these on the `PermissionSet` class as static members instead?

```ts
const perms = new PermissionSet();

// Alternative 1
perms.allow('my-room', PermissionsSet.FULL_ACCESS);

// Alternative 2
perms.allow('my-room', PermissionsSet.presets.FULL_ACCESS);

// Alternative 3
perms.allow('my-room', perms.FULL_ACCESS);
```

Do you have a preference for any of those alternatives, or shall we just keep what we have now?

Reply:
I prefer to ask people to be explicit about it, it's permission, it's best if they know exactly what accesses they are giving.

I prefer alternative 2
`perms.allow('my-room', PermissionsSet.presets.FULL_ACCESS);`

Reply:
I make this argument required in 02fa3a940352834f012d450dde74c0b55ac7df15 👍 
Also, in the latest iteration, the syntax now is:

```ts
const session = liveblocks.prepareSession();
session.allow('my-room', session.FULL_ACCESS);
```

This way, we don't need to import any globals and it's all very easy to work with 👍